aplib.net-demo/Assets/Scripts/MainMenu/CanvasManager.cs

```diff
@@ -73,23 +83,48 @@ private void Start()
         if (_currentSceneName == _sceneNameStartingMenu) // Are we at the starting screen?
         {
             menuCanvas.SetActive(true);
-            settingMenuCanvas.SetActive(false);
-            settingGameCanvas.SetActive(false);
-            _isOnMenuSettings = false;
-            _isOnGameSettings = false;
         }
         else if (_currentSceneName == _sceneNameGame) // Are we at the main gaming scene?
         {
+            _playerHealth = GameObject.FindGameObjectWithTag("Player").GetComponent<HealthComponent>();
+            _playerHealth.Death += OnPlayerDeath;
+
             menuCanvas.SetActive(false);
-            settingMenuCanvas.SetActive(false);
-            settingGameCanvas.SetActive(false);
-            _isOnMenuSettings = false;
-            _isOnGameSettings = false;
         }
         else // TODO:: Remove when game is done. This is for future ease.
         {
             Debug.Log("Scene names are wrong. Check MenuButtons script");
         }
+
+        settingMenuCanvas.SetActive(false);
+        settingGameCanvas.SetActive(false);
+        gameOverCanvas.SetActive(false);
+        _isOnMenuSettings = false;
+        _isOnGameSettings = false;
+    }
+
+    /// <summary>
+    /// When the player dies, the gameover ui is shown.
+    /// </summary>
+    /// <param name="_playerHealth"></param>
+    public void OnPlayerDeath(HealthComponent _playerHealth)
+    {
+        // Mouse visible.
+        UseMouseInSettingsEvent?.Invoke(true);
+
+        // Pause game.
+        Time.timeScale = 0;
+        InputManager.Instance.DisablePlayerInput();
+
+        // Set all off.
+        menuCanvas.SetActive(false);
+        settingMenuCanvas.SetActive(false);
+        settingGameCanvas.SetActive(false);
+        _isOnMenuSettings = false;
+        _isOnGameSettings = false;
```

Comment:
Since you use this code more than once(also in start) maybe extract it to a separate method